listenbrainz/webserver/views/playlist_api.py

```diff
@@ -0,0 +1,516 @@
+
+    try:
+        rows = ujson.loads(r.text)
+    except ValueError as err:
+        current_app.logger.error("Error parse metadata for a playlist: {}".format(err))
+        raise APIInternalServerError("Failed parse fetched metadata for a playlist. Please try again.")
+
+    mbid_index = {}
+    for row in rows:
+        mbid_index[row['original_recording_mbid']] = row
+
+    for rec in playlist.recordings:
+        try:
+            row = mbid_index[str(rec.mbid)]
+        except KeyError:
+            continue
+
+        rec.artist_credit = row.get("artist_credit_name", "")
+        if "[artist_credit_mbids]" in row and not row["[artist_credit_mbids]"] is None:
+            rec.artist_mbids = [UUID(mbid) for mbid in row["[artist_credit_mbids]"]]
+        rec.title = row.get("recording_name", "")
+
+
+@playlist_api_bp.route("/create", methods=["POST", "OPTIONS"])
+@crossdomain(headers="Authorization, Content-Type")
+@ratelimit()
+def create_playlist():
+    """
+    Create a playlist. The playlist must be in JSPF format with MusicBrainz extensions, which is defined
+    here: https://musicbrainz.org/doc/jspf . To create an empty playlist, you can send an empty playlist
+    with only the title field filled out. If you would like to create a playlist populated with recordings,
+    each of the track items in the playlist must have the identifier element present.
+
+    When creating a playlist, only the playlist tite and the track identifier elements will be used -- all
+    other elements in the posted JSPF wil be ignored.
+
+    :reqheader Authorization: Token <user token>
+    :statuscode 200: playlist accepted.
+    :statuscode 400: invalid JSON sent, see error message for details.
+    :statuscode 401: invalid authorization. See error message for details.
+    :statuscode 403: forbidden. The subitting user is not allowed to submit playlists for other users.
+    :resheader Content-Type: *application/json*
+    """
+
+    public = _parse_boolean_arg("public", "true")
+    user = _validate_auth_header()
+
+    data = request.json
+    validate_playlist(data)
+
+
+    playlist = WritablePlaylist(name=data['playlist']['title'], creator_id=user["id"])
+    playlist.public = public
+
+    if data["playlist"].get("created_for", None):
+        if user["musicbrainz_id"] not in current_app.config["APPROVED_PLAYLIST_BOTS"]:
+            raise APIForbidden("Playlist contains a created_for field, but subitting user is not an approved playlist bot.")
+        created_for_user = db_user.get_by_mb_id(data["playlist"]["created_for"])
+        if not created_for_user:
+            log_raise_400("created_for user does not exist.")
+        playlist.created_for_id = created_for_user["id"]
+
+    if "track" in data["playlist"]:
+        for track in data["playlist"]["track"]:
+            playlist.recordings.append(WritablePlaylistRecording(mbid=UUID(track['identifier'][len(PLAYLIST_TRACK_URI_PREFIX):]),
```

Comment:
you do this `[len(PLAYLIST_TRACK_URI_PREFIX):]` here because you've already checked that all items have it in the `validate_playlist` method, right?